### ISSUE TITLE
Serialization for InputMap 

### ISSUE THREAD

**Issue by alradish** [Leafwing-Studios/leafwing-input-manager#279]
## Which feature is frustrating to use or confusing?

For our project, we need to read/write input mappings to/from `.ron` files. 
But serializing prduces really hard to read text. For example, there is no way to know which `Actionlike` type this `InputMap` contains and which row corresponds to which action. And beacuese of this human can't edit this manually.
```Rust
InputMap(
    map: [
        [None,None,None,None,None,None,None,None,None,None,None,None,None,None,None,None],
        [Some(Single(Keyboard(Space))),None,None,None,None,None,None,None,None,None,None,None,None,None,None,None],
        [Some(Single(Keyboard(LShift))),Some(Single(Keyboard(RShift))),None,None,None,None,None,None,None,None,None,None,None,None,None,None],
    ],
    associated_gamepad: None
)
```

## Expectation

It would be great serialize `InputMap` this way, using a custom serde impl for field `map` (or for whole struct).  
```Rust
InputMap(
    map: {
        Hide: [],
        Jump: [Single(Keyboard(Space))],
        Run: [Single(Keyboard(LShift)), Single(Keyboard(RShift))],
    },
    associated_gamepad: None
)
```
In case of using a custom implementation for the entire sturct, we could perhaps extract this behavior to feature, or even deside not serialize field `associated_gamepad` if its `None`.

**Comment by alice-i-cecile** [comment]
I'd prefer a custom implementation for the whole struct, and really like the preferred format.

I'm unsure if associated gamepad should be saved at all: this is session / hardware specific information. Do you think that it can meaningfully be restored?

**Comment by alradish** [comment]
I also thought about the gamepad, but I'm still new to this and have't yet used this part of crate, so I'm not entirely sure.
But if we're fine with not saving the gamepad, then we can maybe implement `From<HashMap<A, Vec<UserInput>>> for InputMap<A>` and use it in custom serialization/deserialization.

**Comment by Shatur** [comment]
I think this could be closed since #282 was merged?
